pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,87 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var desiredReplicas = ptr.Int32(5)
+	var dummyMessage = "test message"
+	var nodeGroupObject *fake.NodeGroup
 	var sng *v1alpha1.ScalableNodeGroup
+	var ng cloudprovider.NodeGroup
+	var defaultNodeGroupObject fake.NodeGroup
 
 	BeforeEach(func() {
 		var err error
+		sng = &v1alpha1.ScalableNodeGroup{}
+		ng = fakeController.CloudProvider.NodeGroupFor(&sng.Spec)
+		defaultNodeGroupObject = *ng.(*fake.NodeGroup)
 		ns, err = env.NewNamespace()
 		Expect(err).NotTo(HaveOccurred())
-		sng = &v1alpha1.ScalableNodeGroup{}
+		sng.Spec.Replicas = desiredReplicas
+		nodeGroupObject = ng.(*fake.NodeGroup)
+	})
+
+	AfterEach(func() {
+		nodeGroupObject.Stable = defaultNodeGroupObject.Stable
```

Comment:
IMO, all this logic should go in a before each

Reply:
There is some duplication I removed that from here. Kept reseting nodeGroupObject, to be specific that we are reseting the nodeGroupObject after every run. Let me know, if you still think it might be better moving to BeforeEach func.